tests/Aspire.EndToEnd.Tests/IntegrationServicesTests.cs

```diff
@@ -1,94 +1,111 @@
-    }
-}
+        if (resourceName == TestResourceNames.cosmos && RuntimeInformation.IsOSPlatform(OSPlatform.OSX) && RuntimeInformation.ProcessArchitecture == Architecture.Arm64)
+        {
+            throw new SkipException($"Skipping '{resourceName}' test because the emulator isn't supported on macOS ARM64.");
+        }
 
-// TODO: remove these attributes when the above tests are running in CI
+        return VerifyComponentWorks(resourceName);
+    }
 
-public class LocalOnlyFactAttribute : FactAttribute
-{
-    public override string Skip
+    [Fact]
+    public async Task KafkaComponentCanProduceAndConsume()
     {
-        get
+        try
         {
-            // BUILD_BUILDID is defined by Azure Dev Ops
+            _integrationServicesFixture.EnsureAppHostRunning();
 
-            if (Environment.GetEnvironmentVariable("BUILD_BUILDID") != null)
-            {
-                return "LocalOnlyFactAttribute tests are not run as part of CI.";
-            }
+            string topic = $"topic-{Guid.NewGuid()}";
 
-            return null!;
+            var response = await _integrationServicesFixture.IntegrationServiceA.HttpGetAsync("http", $"/kafka/produce/{topic}");
+            var responseContent = await response.Content.ReadAsStringAsync();
+            Assert.True(response.IsSuccessStatusCode, responseContent);
+
+            response = await _integrationServicesFixture.IntegrationServiceA.HttpGetAsync("http", $"/kafka/consume/{topic}");
+            responseContent = await response.Content.ReadAsStringAsync();
+            Assert.True(response.IsSuccessStatusCode, responseContent);
+        }
+        catch
+        {
+            _testOutput.WriteLine($"[{DateTime.Now}] <<<< FAILED KafkaComponentCanProduceAndConsume --");
+            await _integrationServicesFixture.DumpDockerInfoAsync();
+            throw;
         }
     }
-}
 
-public class LocalOnlyTheoryAttribute : TheoryAttribute
-{
-    public override string Skip
+    [Fact]
+    public async Task VerifyHealthyOnIntegrationServiceA()
     {
-        get
+        try
         {
-            // BUILD_BUILDID is defined by Azure Dev Ops
+            _integrationServicesFixture.EnsureAppHostRunning();
 
-            if (Environment.GetEnvironmentVariable("BUILD_BUILDID") != null)
-            {
-                return "LocalOnlyTheoryAttribute tests are not run as part of CI.";
-            }
-
-            return null!;
+            // We wait until timeout for the /health endpoint to return successfully. We assume
+            // that components wired up into this project have health checks enabled.
+            await _integrationServicesFixture.IntegrationServiceA.WaitForHealthyStatusAsync("http");
+        }
+        catch
+        {
+            _testOutput.WriteLine($"[{DateTime.Now}] <<<< FAILED VerifyHealthyOnIntegrationServiceA --");
+            await _integrationServicesFixture.DumpDockerInfoAsync();
+            throw;
```

Comment:
Is it possible to do this in a Dispose() method on the class? That way it isn't duplicated across all the tests in the class.

Reply:
Yes, but then it will run for passing tests too. If that's acceptable then I'll move it there.

Reply:
Maybe just refactor the code so it isn't duplicated in all the tests if we can't tell if the test failed or not in Dispose.